src/indexer/segment_writer.rs

```diff
@@ -356,13 +359,22 @@ fn write(
                 .segment()
                 .open_read(SegmentComponent::TempStore)?,
         )?;
+
+        let mut remapped_opstamps = Vec::with_capacity(doc_opstamps.len());
         for old_doc_id in doc_id_map.iter_old_doc_ids() {
+            remapped_opstamps.push(doc_opstamps[*old_doc_id as usize]);
```

Comment:
It would improve readability to separate the computation of the remapped_opstamps from writing the `store_reader`.

Reply:
But that would mean 2 iterations thought ?

Reply:
Yes. I'd do that trade every single day.